python-spec/src/somacore/coordinates.py

```diff
@@ -0,0 +1,491 @@
+"""Definitions of types related to coordinate systems."""
+
+import abc
+import collections.abc
+from typing import Any, Optional, Sequence, Tuple, Union
+
+import numpy as np
+import numpy.typing as npt
+
+
+class Axis:
+    """A description of an axis of a coordinate system
+
+    Args:
+        name: Name of the axis.
+        unit_name: Optional string name for the units. Defaults to ``None``.
+        unit_scale: Optional scale for units. Defaults to ``None``.
```

Comment:
When specifying these with `attrs`, you can put the docstring immediately below the field:

```python
@attrs.define(frozen=True)
class Axis:
  """A description of an axis ..."""

  name: str
  """The name of the axis."""
  unit_name: Optional[str] = None
  """Optional string name ..."""
  # etc.
```